aws/resource_aws_redshift_snapshot_copy_grant.go

```diff
@@ -269,16 +209,5 @@ func findAwsRedshiftSnapshotCopyGrant(conn *redshift.Redshift, grantName string,
 		return nil, err
 	}
 
-	grant := getAwsRedshiftSnapshotCopyGrant(out.SnapshotCopyGrants, grantName)
-	if grant != nil {
-		return grant, nil
-	} else if out.Marker != nil {
-		log.Printf("[DEBUG] Snapshot copy grant not found but marker returned, getting next page via marker: %s", aws.StringValue(out.Marker))
-		return findAwsRedshiftSnapshotCopyGrant(conn, grantName, out.Marker)
-	}
-
-	return nil, &resource.NotFoundError{
-		Message:     fmt.Sprintf("[DEBUG] Grant %s not found", grantName),
-		LastRequest: input,
-	}
+	return out.SnapshotCopyGrants[0], nil
```

Comment:
if err == nil is it guaranteed this slice has at least 1 item?

Reply:
In all my testing that was the case, `err != nil` => `out != nil && len(out.SnapshotCopyGrants)`, but you are correct in that the previous code would have handled opposite by returning `nil` for the grant. It's minor to add that back (plus change some of the callers to check for a `nil` grant).

Reply:
Okay if you don't mind, once that change is made this will get merged! Thank you for helping with all the tech debt.